src/server/services/wmts/qgswmtsutils.cpp

```diff
@@ -152,8 +153,7 @@ namespace QgsWmts
     QgsUnitTypes::DistanceUnit unit = tmi.unit;
 
     // constant
-    double unit_to_m = QgsUnitTypes::fromUnitToUnitFactor( tmi.unit, QgsUnitTypes::DistanceMeters );
-    double resolution = points_to_m * scaleDenominator / unit_to_m;
+    double resolution = tmi.resolution;
```

Comment:
if it's really constant
```suggestion
    const double resolution = tmi.resolution;
```

Reply:
not constant some line after: `resolution = res / 2;`

Reply:
Then better remove the comment ...